hw/templates/snax_accelerator_top_wrapper.sv.tpl

```diff
@@ -0,0 +1,275 @@
+// Copyright 2024 KU Leuven.
+// Solderpad Hardware License, Version 0.51, see LICENSE for details.
+// SPDX-License-Identifier: SHL-0.51
+
+<%
+  num_input_ports = len(cfg["snax_streamer_cfg"]["data_reader_params"]["tcdm_ports_num"])
+  num_output_ports = len(cfg["snax_streamer_cfg"]["data_writer_params"]["tcdm_ports_num"])
+  num_tcdm_ports = num_input_ports + num_output_ports
+
+  # We make the assumption that all reader and writers
+  # Have the same data widths
+  stream_data_width = cfg["snax_streamer_cfg"]["data_reader_params"]["element_width"][0]
```

Comment:
But the output port of the accelerator is larger than the input port for normal accelerators? Is that the cause of the ALU accelerator?

Reply:
After discussing with @xiaoling-yi  we use this:

`stream_data_width = cfg["snax_streamer_cfg"]["fifo_reader_params"]["fifo_width"][0]`